langchain/src/agents/agent.ts

```diff
@@ -195,7 +195,10 @@ export interface AgentArgs {
  * include a variable called "agent_scratchpad" where the agent can put its
  * intermediary work.
  */
-export abstract class Agent extends BaseSingleActionAgent {
+export abstract class Agent<
```

Comment:
Clearly this doesn't work. Basically I wanted to avoid making another layer of inheritance just for `tool` typing